src/utils.rs

```diff
@@ -59,3 +60,15 @@ pub fn bytecode_to_code_hash_for_mode<const N: usize, E: VmEncodingMode<N>>(
 
     Ok(versioned_hash_bytes)
 }
+
+/// Erase start and page number from a fat pointer. To be used in the case of a fat pointer
+/// being passed to an opcode which shouldn't receive one.
+pub fn erase_fat_pointer_metadata(ptr: &mut U256) {
+    // Memory page is at 1<<32 - 1<<64, start is at 1<<64 - 1<<96
+    // We also need to preserve the top 128 bits of the value
+    let low_bits = 0xffffffff0000000000000000ffffffffu128;
```

Comment:
you can format it for readability (by 32-bit chunks), or even better you can access U256 internal representation as `ptr.0` with u64 words inside, that is more efficient that computing full 256 bit binop